badges/models.py

```
@@ -0,0 +1,342 @@
+        constraints = [
+            models.UniqueConstraint(
+                fields=[
+                    "user",
+                    "achievement",
+                    "source_content_type",
+                    "source_object_id",
+                ],
+                condition=models.Q(source_type="automatic"),
+                name="unique_automatic_user_achievement_source",
+            )
+        ]
```

Ruff (0.16.0)

[warning] Mutable default value for class attribute

(RUF012)